source/core/src/main/com/csse3200/game/components/player/PlayerStatsDisplay.java

```diff
@@ -58,12 +57,42 @@ public void create() {
 
         entity.getEvents().addListener("updateHealth", this::updatePlayerHealthUI);
         entity.getEvents().addListener("melee_pickup", this::updateMeleeWeaponUI);
-        entity.getEvents().addListener("ranged_pickup", this::updateRangedWeaponUI);
+        entity.getEvents().addListener("ranged_pickup", (Integer maxAmmo) -> {
+            updateGunLabel(20, "Gun");
+            updateRangedWeaponUI(maxAmmo, "Gun");
+        });
         entity.getEvents().addListener("ranged_activate", this::updateAmmoDisplay);
         entity.getEvents().addListener("updateSpeedPercentage", this::updateSpeedPercentageUI);
         entity.getEvents().addListener("updateDamageBuff", this::updateDamageUI);
         entity.getEvents().addListener("updateSpeedUI", this::updateSpeedPercentageUI);
         entity.getEvents().addListener("updateArmor", this::updateArmorUI);
+        initializeWeaponsAndAmmo();
+    }
+
+    /**
+     * Initializes the weapon and ammo counts based on the player selected.
+     * Retrieves the player type from PlayerAnimationController.
+     */
+    private void initializeWeaponsAndAmmo() {
+        // Retrieve the PlayerAnimationController component from the entity
+        PlayerAnimationController playerController = entity.getComponent(PlayerAnimationController.class);
+
+        // Get the current player number
+        PlayerAnimationController.PlayerNum playerNum = playerController.getPlayerNum();
+
+        // Based on the selected player, display appropriate weapon and ammo counts
+        if (playerNum == PlayerAnimationController.PlayerNum.PLAYER_4 ||
+                playerNum == PlayerAnimationController.PlayerNum.PLAYER_2 ||
+                playerNum == PlayerAnimationController.PlayerNum.PLAYER_3 ||
+                playerNum == PlayerAnimationController.PlayerNum.BEAR) {
+            // Players 2, 3, 4, and Bear have 7 ammo with the gun
+            updateGunLabel(7, "Gun");
+            updateRangedWeaponUI(7, "Gun");
+        } else {
+            // For other players, set the gun to have 20 ammo
+            updateGunLabel(20, "Gun");
+            updateRangedWeaponUI(20, "Gun");
```

Comment:
the padding hasn't been configured properly. it is making the goblin statue render improperly:
![image](https://github.com/user-attachments/assets/590ddd4a-2f1f-427f-9345-8f69411488df)